### PR TITLE
fix: renaming folder in sidebar doesnt update Route

### PR DESCRIPTION
**Pull request by jyash97**
## What kind of change does this PR introduce?
Bug Fix

## What is the current behavior?
When we rename a folder in Dashboard Sandboxes, it doesn't update the route. 

Folder_Renaming_Error

## What is the new behavior?
Update the route when the folder is renamed

Renaming_Error_Fixed

## What steps did you take to test this? This is required before we can merge, make sure to test the flow you've updated.

1. Tested on local check the above GIF for more details

## Checklist

- [ ] Documentation
- [x] Testing 
- [x] Ready to be merged
      
- [x] Added myself to contributors table
      

I was seeing a `TypeScript` error on `DropFolderEntry`  and I am not sure how to fix that. May be CI pipeline will fail for that. 

image

Any help is appreciated on this!

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/2956/commits/0fc3e0cf23312955368c0153c67a4629d37ee0cc is at https://pr2956.build.csb.dev/s/new.

File: packages/app/src/app/pages/Dashboard/Sidebar/SandboxesItem/FolderEntry/index.tsx [hunk 1]
```diff
@@ -54,6 +55,9 @@ type Props = {
   currentPath: string;
   currentTeamId: string;
 
+  // router props
```
Comment (CompuIves) [inline packages/app/src/app/pages/Dashboard/Sidebar/SandboxesItem/FolderEntry/index.tsx]: We can add these props by doing

```
type Props = { .. } & RouteComponentProps;
```

Then we have it properly typed as well 😄

Comment (jyash97) [reply packages/app/src/app/pages/Dashboard/Sidebar/SandboxesItem/FolderEntry/index.tsx]: Hey, thanks for the quick help.  But this doesn't seem to work. Any ideas why?

Comment (CompuIves) [reply packages/app/src/app/pages/Dashboard/Sidebar/SandboxesItem/FolderEntry/index.tsx]: I'll take a look!

**Comment by CompuIves** [comment]
Oh boy, it feels like I opened a can of worms with this. I went with the easy solution in the end, let's see if this gets the green light and merge it then 😄

**Comment by jyash97** [comment]
Hey thanks for taking a look into this :+1: 
I can still see 1 test failing but I guess that is a separate issue and not related to this.

**Comment by CompuIves** [comment]
Yep, thanks a lot! This looks good!

**Pull request merged**